### PR TITLE
temp fix for Android

### PR DESCRIPTION
**Pull request by sandstranger**

### CODE REVIEW

**Comment by scrawl** [comment]
Not acceptable. Instead we need to find out why the delete function doesn't work.

**Comment by sandstranger** [comment]
This is madness , but openmw crashed if declared more than one bools types in refdata.hpp

**Comment by scrawl** [comment]
Sounds like an alignment bug. Reordering the bools might work around it, i.e. declare the 4 bools after one another in the header.

**Comment by sandstranger** [comment]
Please not  merge this commit  to the repository,I found more bugs with bools, which returns wrong values..

**Comment by sandstranger** [comment]
Please merge this commit to the repository,  I will deal with graphics later . 
Now I need start refactor Java code to make it work without bugs

File: apps/openmw/mwmechanics/creaturestats.hpp [hunk 1]
```diff
@@ -33,6 +33,23 @@ namespace MWMechanics
         MagicEffects mMagicEffects;
         Stat<int> mAiSettings[4];
         AiSequence mAiSequence;
+#ifdef ANDROID
```
Comment (Mingun) [inline apps/openmw/mwmechanics/creaturestats.hpp]: The order of fields is important? Why not simple reorganize them as you made for Android, for all platforms and delete `#ifdef`?

File: apps/openmw/mwworld/refdata.hpp [hunk 2]
```diff
@@ -33,12 +33,17 @@ namespace MWWorld
             MWScript::Locals mLocals; // if we find the overhead of heaving a locals
                                       // object in the refdata of refs without a script,
                                       // we can make this a pointer later.
+#ifdef ANDROID
```
Comment (Mingun) [inline apps/openmw/mwworld/refdata.hpp]: Also as above

File: apps/openmw/mwmechanics/creaturestats.hpp [hunk 3]
```diff
@@ -33,19 +33,22 @@ namespace MWMechanics
...
+        mAlarmed,
+        mAttacked,
+        mAttackingOrSpell,
+        mKnockdown,
+        mKnockdownOneFrame,
+        mKnockdownOverOneFrame,
+        mHitRecovery,
+        mBlock,
+        mRecalcMagicka;
         int mFriendlyHits;
```
Comment (zinnschlag) [inline apps/openmw/mwmechanics/creaturestats.hpp]: Can we please not do that? One separate declaration per variable (as we had before).

**Comment by sandstranger** [comment]
It is done

**Comment by zinnschlag** [comment]
Sorry, but that does not look good. These two pieces of code should be equivalent. Are we actually talking about a compiler bug here? I think further investigations are required regarding why the clean version of this piece of code causes problems on Android.

**Comment by sandstranger** [comment]
Ok ,I will try to deal with it , perhaps the problem was in the flags gcc, which I used in the building

**Comment by sandstranger** [comment]
game crashed here due to use gcc flags: -O3 ,-O2 or -Ofast

**Comment by sandstranger** [comment]
Even without -O2, -O3, -Ofast game crashes after start loading game .
but if change mDeleted like this :
            bool mDeleted; // separate delete flag used for deletion by a content file
            bool mHasLocals;
            bool mEnabled;
            int mCount; // 0: deleted
game stop crashes

**Comment by zinnschlag** [comment]
Closed or ready to merge?

**Comment by sandstranger** [comment]
ready to merge I think

**Comment by kcat** [comment]
Did you ever figure out what's wrong? As far as I can see, the fixes are still just reordering some fields. Since that makes no sense, hiding the problem with random changes feels like the wrong thing to do if we want it stable. Plus, these 'add ifdef'/'delete ifdefs'  commits make the log a bit messy since they're just reverting what the patchset just did. Would it be possible to do a rebase without those commits?

**Comment by sandstranger** [comment]
Ok , than I will delete it . Problem with crashing game connected with mDeleted . How to fix it with another way I do not know yet

**Pull request closed without merging**